dockerfiles/ruby-3.3.Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM ruby:3.3-alpine
+
+RUN apk add --no-cache --upgrade 'curl>=7.66'
```

Comment:
Hmm, I don't think we need this - not sure why the older dockerfile has it. 

We handle installing `curl` here: https://github.com/codecrafters-io/core/blob/674d9e5ab7a41ab95e97003c9a01da55a4b85a78/app/lib/buildpack_dockerfile_processor.rb#L29 (that's how all the other alpine images work without this line)